### DISCUSSION TITLE
Tenant is not set when using createOptionForm()

### DISCUSSION THREAD

**Discussion by axelbdt** [filamentphp/filament#8254] [Help]
Hello,

I have a Booking resource and a Client resource that both belong to my tenant model Company. Booking belongs to the model, so the form has a Selection field to select clients, and I used createOptionForm() to be able to create Client on the fly while filling informations for a Booking.

```php
// In my BookingResource class
public static function form(Form $form): Form
    {
        return $form
            ->schema(
                [
                Forms\Components\Select::make('client_id')
                    ->label(__('client'))
                    ->required()
                    ->relationship('client', 'name')
                    ->searchable()
                    ->preload()
                    ->disablePlaceholderSelection()
                    ->createOptionForm(
                        ClientResource::form_schema()
                    ),
                   //...
            ]
}
```

Creating a Client on the dedicated resource works, but when using the modal from the Booking form, the company_id is null and breaks constraints.

Can you indicate a way to set the company_id field? Should the tenant be set automatically when using a createOptionForm?

Cheers

**Comment by danharrin** [comment]
> Should the tenant be set automatically when using a createOptionForm?

No, as mentioned in the Tenancy documentation we do not touch form components, we only associate resources with tenants

You can try this, or use an observer

```php
->createOptionAction(function ($action) {
    $action->mutateFormData(function ($data) {
        $data['company_id'] = Filament::getTenant()->id;

        return $data;
    });
})
```

**Reply by BenKruft** [reply]
Thank you! This works with:
`mutateFormDataUsing`

Now i just need to find out how i can auto-select the newly generated option :)

**Reply by danharrin** [reply]
You return the ID of the new option from the action
